OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -165,6 +165,24 @@ public GameInfoStatsLogic(Widget widget, ModData modData, World world, OrderMana
 					muteCheckbox.IsVisible = () => !pp.IsBot && client.State != Session.ClientState.Disconnected && pp.ClientIndex != orderManager.LocalClient?.Index;
 					muteCheckbox.GetTooltipText = () => muteCheckbox.IsChecked() ? unmuteTooltip : muteTooltip;
 
+					var kickButton = item.Get<ButtonWidget>("KICK");
+					kickButton.IsVisible = () => Game.IsHost && client.Index != orderManager.LocalClient?.Index && client.State != Session.ClientState.Disconnected && pp.WinState == WinState.Lost && !pp.IsBot;
+					kickButton.OnClick = () =>
```

Comment:
Can we share this logic with the kick button for spectators? (It seems to be identical.)

Reply:
abstracted